ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,559 @@
+        if spanning_tree_root_super is True or spanning_tree_mode is not None:
+            spanning_tree = {}
+            if spanning_tree_root_super is True:
+                spanning_tree.update({'root_super': True})
+            if spanning_tree_mode is not None:
+                spanning_tree.update({'mode': spanning_tree_mode})
+                if spanning_tree_mode == "mstp":
+                    spanning_tree.update({'mst_instances': {"0": {'priority': get(self._switch_data,
+                                                                                  'spanning_tree_priority', '32768')}}})
+                elif spanning_tree_mode == "rapid-pvst":
+                    spanning_tree.update({'rapid_pvst_instances':
+                                         {"1-4094": {'priority': get(self._switch_data, 'spanning_tree_priority',
+                                                                     '32768')}}})
+                elif spanning_tree_mode == "rstp":
+                    spanning_tree.update({'rstp_priority': get(self._switch_data, 'spanning_tree_priority', '32768')})
+            return spanning_tree
+        return None
+
+    @cached_property
+    def service_unsupported_transceiver(self):
+        unsupported_transceiver = get(self._hostvars, 'unsupported_transceiver')
+        if unsupported_transceiver is not None:
+            return {'license_name': get(unsupported_transceiver, 'license_name'),
+                    'license_key': get(unsupported_transceiver, 'license_key')}
+        return None
+
+    @cached_property
+    def local_users(self):
+        local_users = get(self._hostvars, 'local_users')
+        if local_users is not None:
+            local_users = self._convert_dicts(local_users, 'name')
+            local_users_dict = {}
+            for local_user in self._natural_sort(local_users, 'name'):
+                name = get(local_user, 'name')
+                local_users_dict.update({name: {'privilege': get(local_user, 'privilege')}})
+                role = get(local_user, 'role')
+                sha512_password = get(local_user, 'sha512_password')
+                no_password = get(local_user, 'no_password')
+                ssh_key = get(local_user, 'ssh_key')
+                if role is not None:
+                    local_users_dict[name].update({'role': role})
+                if sha512_password is not None:
+                    local_users_dict[name].update({'sha512_password': sha512_password})
+                elif no_password is not None:
+                    local_users_dict[name].update({'no_password': no_password})
+                if ssh_key is not None:
+                    local_users_dict[name].update({'ssh_key': ssh_key})
+            return local_users_dict
+        return None
+
+    @cached_property
+    def clock(self):
+        timezone = get(self._hostvars, 'timezone')
+        if timezone is not None:
+            return {'timezone': timezone}
+        return None
```

Comment:
@ClausHolbechArista at the end do we remove vairables like `{'timezone': None}`? Just wondering if we could just replace this with

```
return {'timezone': get(self._hostvars, 'timezone')}
```
and let the rest of the code deal with it.

Reply:
No we will not "dive in" to the returned data models. Only check the outermost key. This is a slight deviation from the previous YAML method, but is much simpler and quicker. Also it was required to skip the "remove_empties" stuff from the python area because of some old issues/bugs.